cardinal/storage/newstorage/kv.go

```diff
@@ -0,0 +1,48 @@
+package newstorage
+
+// KVStore is a generic key-value store interface with batch operations.
+type KVStore[K comparable, V any] interface {
```

Comment:
You would use a single KVStore for both EntityID and Component here. From the abstraction's perspective, I don't think it makes sense to expect atomicity across what could be backed by a different DB to begin with

Reply:
There are stores that can hold multiple types. KVStore forces it to hold a single type (unless you use any). Thus not all DBs can be fully represented by a single KVStore. They have to use multiple KVStores. These DBs have atomicity across different types and this atomicity is harder to represent via multiple KVStores.

Reply:
Database technically only operates on a single data type, []byte. types can't be transported over the network or the file system so they always need to be serialized.

Sure the database might have schema data types, but those are transported as a stream of bytes that is then typecasted.

My point for saying is that you can just use 'any' as the value data type if you really want to avoid serialization and that would comparable to how DBs strictly only operate on byte streams I/O

Reply:
Per scott's original comment of using the single KVStore for both EntityID and Component, we would have this (terrible name nonwithstanding):

```
type ECBImpl struct {
  EntityIDAndComponentAndEtcStore KVStore[string, any]
}
```

If we plan on using KVStore in other places (e.g. nonce tracking and schema tracking), ok. The value type will be something else. If this KVStore is only going to be used in storage/ECB, it's not at at clear what the generic parameters actually get us. We should just define the KVStore interface as having `string` keys and `any` values.